optaplanner-core/src/main/java/org/optaplanner/core/impl/score/stream/common/AbstractJoiner.java

```diff
@@ -18,14 +18,6 @@
 
 public abstract class AbstractJoiner {
 
-    protected void assertMappingIndex(int index) {
-        int mappingCount = getJoinerTypes().length;
-        if (index >= mappingCount) {
-            throw new IllegalArgumentException(
-                    "Joiner only has (" + mappingCount + ") mappings, requested index (" + index + ")");
-        }
-    }
-
```

Comment:
I am removing this as it just replaces an exception with another one.
This brings no value and costs precious time on the CS-D hot path.

Reply:
Agreed.